api/hiveextension/v1beta1/agentclusterinstall_types.go

```diff
@@ -387,6 +387,9 @@ const (
 
 	// NutanixPlatformType represents Nutanix infrastructure.
 	NutanixPlatformType PlatformType = "Nutanix"
+
+	// ExternalPlatformType represents external cloud provider infrastructure.
+	ExternalPlatformType PlatformType = "External"
```

Comment:
line 375 should be updated as well, no ?

Reply:
What are the problems with adding it here?

Reply:
Also do we need a [provider implementation](https://github.com/openshift/assisted-service/tree/ba04805cce13bf9b32647539c33ee63f85831586/internal/provider) for this?

Reply:
@adriengentil did we add external platform support to assisted-service without adding it to the k8s API? Maybe I'm not understanding the purpose of this PR.

Reply:
@mhrivnak This PR is basically to help with the supported platform types for agent installer Ref https://github.com/openshift/installer/pull/7442/files#diff-fbe7c3bdbc1d6bab9c543d30777d3955cd42872e4eb9c4f592f6137a6da5472dR40

Reply:
> What are the problems with adding it here?
None 🙂 For some reason, I checked https://github.com/openshift/hive/tree/master/apis/hive/v1 and questioned about it initially.

Reply:
@pawanpinjarkar I get that, but I'm not clear on why this specific change is what's needed in order to "help with the supported platform types for agent installer".

Making some assumptions, I suppose that agent-installer is using the k8s API structures for input, and assisted-service may have neglected to add external platform support to its k8s API? If that's correct, then presumably more code is required in assisted-service to enable this in the k8s API than just adding another enum value.

Or did assisted-service enable it differently in the k8s API?

Reply:
I did nothing related to the k8s API, I discussed with @pawanpinjarkar and from what I understand we'll need to implement a new platform type "external" in AI (through a new provider, as @carbonin said), then we'll be able to plumb it in the k8s API.

I'll try to hack something today, it shouldn't be very complex as most of the pieces are already in place.

Reply:
I'm not sure that this change is enough, looking at how nutanix support was added https://github.com/openshift/assisted-service/pull/4996 
you will need to generate the CRs post api changes. 
Because it's in the api it will mean that k8s api need to support that and lets not forget that we are GA so if someone will use that they will expect it to work. 
In the service is there a support for external platform? i suspect that not, based on https://github.com/openshift/assisted-service/blob/8534d1ece9dbd935a40436b69c1dfd5322ff9a09/swagger.yaml#L6221-L6228

Reply:
> I did nothing related to the k8s API, I discussed with @pawanpinjarkar and from what I understand we'll need to implement a new platform type "external" in AI (through a new provider, as @carbonin said), then we'll be able to plumb it in the k8s API.
> 
> I'll try to hack something today, it shouldn't be very complex as most of the pieces are already in place.

Regarding the external platform, as assisted service expects the platformName to be `oci`, we in the agent installer, for 4.14, will add validation at the install-config level to only set `platformName` to `oci` if platform is `external` that way, the generic implementation of the platformName can be deferred until 4.15

Reply:
e.g. https://github.com/openshift/installer/pull/7442/files#diff-0becd9dba8c3961ae565ce49accf81a33fdd6e513de100c0cc9ac7f55650ddc9R3

Reply:
We're going to have a huddle on the assisted-service side to make sure we're providing the right "generic" external platform API for you to use. I know it's time-sensitive, but due to PTOs, it'll be Monday before we can have a full conversation. But I wanted to make sure you're in the loop on that intent. @adriengentil is working on it in the meantime as well.

Reply:
@zaneb Once the generic external platform is available, should the agent installer then just accept `external` or we can also accept `oci` along with `external`?

Reply:
> We're going to have a huddle on the assisted-service side to make sure we're providing the right "generic" external platform API for you to use. I know it's time-sensitive, but due to PTOs, it'll be Monday before we can have a full conversation. But I wanted to make sure you're in the loop on that intent. @adriengentil is working on it in the meantime as well.

@mhrivnak Please keep us posted. As of today, I believe we have everything up for review from the agent installer side. With a new PR in assisted-image-service, we are now able to generate the minimal ISO as well.

Reply:
Got the necessary info from slack thread discussions.